tapable.d.ts

```diff
@@ -86,12 +156,38 @@ declare class AsyncHook<T, R, AdditionalOptions = UnsetAdditionalOptions> extend
 	): void;
 }
 
-export class AsyncParallelHook<T, AdditionalOptions = UnsetAdditionalOptions> extends AsyncHook<T, void, AdditionalOptions> {}
-export class AsyncParallelBailHook<T, R, AdditionalOptions = UnsetAdditionalOptions> extends AsyncHook<T, R, AdditionalOptions> {}
-export class AsyncSeriesHook<T, AdditionalOptions = UnsetAdditionalOptions> extends AsyncHook<T, void, AdditionalOptions> {}
-export class AsyncSeriesBailHook<T, R, AdditionalOptions = UnsetAdditionalOptions> extends AsyncHook<T, R, AdditionalOptions> {}
-export class AsyncSeriesLoopHook<T, AdditionalOptions = UnsetAdditionalOptions> extends AsyncHook<T, void, AdditionalOptions> {}
-export class AsyncSeriesWaterfallHook<T, AdditionalOptions = UnsetAdditionalOptions> extends AsyncHook<T, AsArray<T>[0], AdditionalOptions> {}
+export class AsyncParallelHook<
+	T,
+	AdditionalOptions = UnsetAdditionalOptions,
+	ContextType = unknown
+> extends AsyncHook<T, void, AdditionalOptions, ContextType> {}
+export class AsyncParallelBailHook<
+	T,
+	R,
+	AdditionalOptions = UnsetAdditionalOptions,
+	ContextType = unknown
```

Comment:
Why it is unknown?

Reply:
I wanted to avoid the use of `any` if possible. Context starts as an empty object, so I could use something like `Record<any, unknown>`